hr_expense_sheet_payment_state/models/account_move.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def is_invoice(self, include_receipts=False):
+        # Override move type check in the context of payment_state computation,
+        # because the move linked to an expense sheet is of type entry.
+        self.ensure_one()
+        if self.line_ids.expense_id:
```

Comment:
Actually looks like this is already possible in v14.0 (I also thought it was v.15 only)

Reply:
ok I didn't know about that, even so on implementation for v15 this check is done in a separate hook and method is_invoice is not overridden completely, this is not very safe
anyhow build should be fixed

Reply:
About the build.
We have an error on field which isn't installed to db by our module.
All the modules are installed during test instance is prepared.
I tried to run tests specifying different tags to tagged decorator to avoid testing on the fields which aren't related to my module, but still has an error. As it's not related to my module, we tested locally with @grindtildeath and left it as it's.

Reply:
@yankinmax I guess to make this safer, you could add a context key on the `_compute_amount` function since both calls to `is_invoice` were replaced by calls to `_payment_state_matters`, and check the existence of the context key before returning `True`.

Reply:
@i-vyshnevska does it look safer now?

Reply:
yes, thanks